tests/gcd.rs

```diff
@@ -0,0 +1,101 @@
+use core::fmt::Debug;
+use core::ops::{Add, Mul};
+use ibig::{
+    ops::{ExtendedGcd, Gcd},
+    ubig,
+};
+
+fn test_gcd<'a, T, U>(a: &'a T, b: &'a T, c: &'a T)
+where
+    T: Gcd<T, Output = T> + ExtendedGcd<T, OutputGcd = T, OutputCoeff = U>,
+    T: Gcd<&'a T, Output = T> + ExtendedGcd<&'a T, OutputGcd = T, OutputCoeff = U>,
+    &'a T: Gcd<T, Output = T> + ExtendedGcd<T, OutputGcd = T, OutputCoeff = U>,
+    &'a T: Gcd<&'a T, Output = T> + ExtendedGcd<&'a T, OutputGcd = T, OutputCoeff = U>,
+    T: Clone + Debug + Eq,
+    U: From<T> + Clone + Debug + Eq,
+    U: Add<U, Output = U> + Mul<U, Output = U>,
+{
+    assert_eq!(a.gcd(b), *c);
+    assert_eq!(a.clone().gcd(b), *c);
+    assert_eq!(a.gcd(b.clone()), *c);
+    assert_eq!(a.clone().gcd(b.clone()), *c);
+
+    let (g, x, y) = a.extended_gcd(b);
+    assert_eq!(&g, c);
+    assert_eq!(x * U::from(a.clone()) + y * U::from(b.clone()), U::from(g));
+
+    let (g, x, y) = a.extended_gcd(b.clone());
+    assert_eq!(&g, c);
+    assert_eq!(x * U::from(a.clone()) + y * U::from(b.clone()), U::from(g));
+}
+
+#[test]
+fn test_gcd_ubig() {
+    // test cases (x, y, gcd(x,y))
+    let test_cases = [
+        // trivial cases
+        (ubig!(0), ubig!(0), ubig!(0)),
```

Comment:
gcd(0, 0) should be a panic

Reply:
Done in https://github.com/tczajka/ibig-rs/pull/11/commits/c618f14544328e83626bac2a70fc4d2af10d2b29